sig-architecture/production-readiness.md

```diff
@@ -41,6 +41,62 @@ aspects of the process.
   with and without the feature are necessary. At the very least, think about
   conversion tests if API types are being modified.
 
+
+#### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+* **What signals should trigger rollback?**
+  Focus on metrics and treat other signals (like logs) as last resort.
+
+* **Were upgrade and rollback tested?**
```

Comment:
upgrade, downgrade, upgrade.  We've had.... issues.

Reply:
added